main.tf

```diff
@@ -347,8 +353,11 @@ module "rds_identifier" {
 }
 
 resource "aws_rds_cluster_instance" "default" {
-  count                                 = local.cluster_instance_count
-  identifier                            = "${module.rds_identifier[0].id}-${count.index + 1}"
+  count = local.cluster_instance_count
+
+  identifier_prefix = var.rds_cluster_identifier_prefix_enabled ? "${module.rds_identifier[0].id}${module.this.delimiter}${count.index + 1}${module.this.delimiter}" : null
```

Comment:
The unique name for the aws_rds_cluster_instance should be controlled by the random_pet that feeds into the module.rds_identifier. Why is that not working here?

Reply:
How do you know it's not working?

Reply:
Oh I see what you're saying. The random pet does go into the rds_identifier. Hmm, maybe I'm not using it correctly

Reply:
@nitrocode there is some history here, I made some changes in  #236 but after some FUD some of the changes were taken out in #242. You would probably need to add cluster_identifier back into the random_pet to get the features you want.